src/libs/geotagging.c

```diff
@@ -863,23 +1679,132 @@ void gui_init(dt_lib_module_t *self)
+  g_signal_connect(G_OBJECT(d->map.gpx_button), "clicked", G_CALLBACK(_choose_gpx_callback), self);
+
+  d->map.gpx_file = dt_ui_label_new("");
+  gtk_label_set_ellipsize(GTK_LABEL(d->map.gpx_file ), PANGO_ELLIPSIZE_MIDDLE);
+  gtk_widget_set_hexpand(d->map.gpx_file, TRUE);
+  gtk_grid_attach(grid, d->map.gpx_file, 1, line++, 3, 1);
+  gtk_box_pack_start(GTK_BOX(d->map.gpx_section), GTK_WIDGET(grid), TRUE, TRUE, 0);
+
+  model = gtk_list_store_new(DT_GEO_TRACKS_NUM_COLS, G_TYPE_BOOLEAN, G_TYPE_STRING,
+                             G_TYPE_INT, G_TYPE_INT, G_TYPE_INT, G_TYPE_STRING);
+
+  d->map.gpx_view = gtk_tree_view_new_with_model(GTK_TREE_MODEL(model));
+  g_object_unref(model);
+  gtk_widget_set_name(d->map.gpx_view, "gpx_list");
+  gtk_widget_set_tooltip_text(GTK_WIDGET(d->map.gpx_view),
+                              _("list of track segments in the gpx file, for each segment:"
+                                "\n- the start date time in local time (LT)"
+                                "\n- the number of track points"
+                                "\n- the number of images going to be geotagged "
+                                "based on images date time, offset and time zone"
+                                "\n- more detailled time information hovering the row"));
+  renderer = gtk_cell_renderer_toggle_new();
+  g_signal_connect(renderer, "toggled", G_CALLBACK(_track_seg_toggled), self);
+  GtkTreeViewColumn *column = gtk_tree_view_column_new_with_attributes("", renderer, "active", DT_GEO_TRACKS_ACTIVE, NULL);
+  gtk_tree_view_append_column(GTK_TREE_VIEW(d->map.gpx_view), column);
+  d->map.sel_tracks = column;
+  GtkWidget *button = gtk_check_button_new();
+  gtk_widget_show(button);
+  gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(button), FALSE);
+  gtk_tree_view_column_set_widget(column, button);
+  gtk_tree_view_column_set_alignment(column, 0.5);
+  g_signal_connect(column, "clicked", G_CALLBACK(_all_tracks_toggled), self);
+
+  column = _new_tree_text_column(_("start time"), TRUE, 0.0, DT_GEO_TRACKS_DATETIME, PANGO_ELLIPSIZE_START);
+  gtk_tree_view_append_column(GTK_TREE_VIEW(d->map.gpx_view), column);
+  column = _new_tree_text_column(_("points"), FALSE, 1.0, DT_GEO_TRACKS_POINTS, PANGO_ELLIPSIZE_NONE);
+  gtk_tree_view_append_column(GTK_TREE_VIEW(d->map.gpx_view), column);
+  column = _new_tree_text_column(_("images"), FALSE, 1.0, DT_GEO_TRACKS_IMAGES, PANGO_ELLIPSIZE_NONE);
+  gtk_tree_view_append_column(GTK_TREE_VIEW(d->map.gpx_view), column);
+
+  g_object_set(G_OBJECT(d->map.gpx_view), "has-tooltip", TRUE, NULL);
+  g_signal_connect(G_OBJECT(d->map.gpx_view), "query-tooltip", G_CALLBACK(_row_tooltip_setup), self);
+
+  // avoid ugly console pixman messages due to headers
+  gtk_tree_view_set_headers_visible(GTK_TREE_VIEW(d->map.gpx_view), FALSE);
+  GtkWidget *w = dt_ui_scroll_wrap(GTK_WIDGET(d->map.gpx_view), 100, "plugins/lighttable/geotagging/heighttracklist");
```

Comment:
The initial track list widget height is very small, less than one line. You need to add a preference into darktableconfig.xml.in with a minimal height value.